### PR TITLE
Implement disk creation

### LINKED ISSUE #218: Implement Linode detail settings Advanced Disk add a disk

### PR DESCRIPTION
**Pull request by SirCmpwn**
Closes #218

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Error handling:

File: scss/components/modal.scss [hunk 1]
```diff
@@ -16,8 +16,8 @@
 .modal {
     position: absolute;
     left: 50%;
-    width: 400px;
-    margin-left: -200px;
+    width: 500px;
+    margin-left: -250px;
```
Comment (eatonphil) [inline scss/components/modal.scss]: We'll be getting rid of this stuff soon enough. Initially I was just doing the task of col-*-* but for only one size. Eventually we'll get rid of the widths here and use col-xs, col-sm, ... appropriately.

File: src/linodes/settings/components/DiskPanel.js [hunk 2]
```diff
@@ -237,6 +462,8 @@ export class DiskPanel extends Component {
                   <p>{free} MiB</p>
                   {poweredOff ?
                     <button
+                      onClick={() => dispatch(showModal('Add a disk',
+                        <AddModalRedux free={free} linode={linode} />))}
```
Comment (eatonphil) [inline src/linodes/settings/components/DiskPanel.js]: Didn't notice this sort of thing before. Do you like this? (Passing components to an action constructor)

Comment (SirCmpwn) [inline src/linodes/settings/components/DiskPanel.js]: Sure, why not? We already do this everywhere else showModal is used, only now it's formalized as actual components.

Comment (eatonphil) [inline src/linodes/settings/components/DiskPanel.js]: Nvm. Where I was going with this was mentioning having multiple modal instances in the layout. Don't think I want to go there though.

Comment (SirCmpwn) [inline src/linodes/settings/components/DiskPanel.js]: Yeah, let's not go there.

**Comment by eatonphil** [comment]
This looks really good. I'm really happy with the work you're doing on this page visually. Gunna log into the CF Manager to see if we're missing anything else relevant. Is this still [WIP]?

**Comment by SirCmpwn** [comment]
Nope, just forgot to remove it.

**Comment by eatonphil** [comment]
I guess the one aspect of the deploy to disk flow this is glossing over is the swap. But I guess since this is post-create it doesn't matter in the same way. Actually, this makes me think the distribution option should come after the disk type. If you select swap, the distribution should be hidden. And is there a root password option? How do they set that for their new disk with a distribution?

**Comment by SirCmpwn** [comment]
Root pw appears after you select a distro (and fs disappears):

**Comment by eatonphil** [comment]
Why does fs disappear? -1. I think the fs should be above the distro and always be visible. Then distro should be hidden if they select swap (or raw, actually) as the fs.

**Comment by SirCmpwn** [comment]
You can't pick the fs if you are asking us to deploy a distro. It comes with the distro image.

**Comment by eatonphil** [comment]
That is... awkward. We only have five fs options. ext3, ext4, raw, swap, and initrd. If the user selects ext3 or ext4 then the distros are available to them. Otherwise hide the distros. Are all distros not available on both ext3 and ext4?

**Comment by SirCmpwn** [comment]
-1

**Comment by SirCmpwn** [comment]
Your choice between ext3/4 (or any other kind of fs) is discarded by the host and it just deploys whatever it feels like when you choose a distro.

**Comment by SirCmpwn** [comment]
Why leave the UI there for a meaningless choice once you've selected a distro?

**Comment by eatonphil** [comment]
Maybe this is the wrong UI here then. Presenting it linearly here suggests any combination of what they see is an option. The option they really have is: deploy distribution or select a blank filesystem. (And we're not even talking about StackScripts which will have to be an option here too later.)

**Comment by SirCmpwn** [comment]
I think the UI effectively conveys the flow you just described. When you get to the distro choice, if you pick none, you're offered a filesystem choice. If you pick one, you're asked for a root pw.

**Comment by eatonphil** [comment]
It does not convey that no distribution is deployable to a raw or swap fs. Since the option is still there once they'd selected raw, it would seem like they could then go back and select a distro.

**Comment by SirCmpwn** [comment]
If the user goes out of order it's not my fault.

**Pull request merged**